lib/cog/command/pipeline/executor.ex

```diff
@@ -231,20 +231,11 @@ defmodule Cog.Command.Pipeline.Executor do
         resp = Cog.Command.Response.decode!(payload)
         case resp.status do
           "ok" ->
-            collected_output = case resp.body do
-                                 nil ->
-                                   # If there wasn't any output,
-                                   # there's nothing to collect
-                                   state.output
-                                 body ->
-                                   state.output ++ Enum.map(List.wrap(body),
-                                                            &store_with_template(&1, resp.template))
-                                   # body may be a map or a list
-                                   # of maps; if the latter, we
-                                   # want to accumulate it into
-                                   # one flat list
-                               end
+            collected_output = collect_output(resp, state.output)
             {:next_state, :execute_plan, %{state | output: collected_output}, 0}
+          "abort" ->
+            collected_output = collect_output(resp, state.output)
```

Comment:
I tried to be conservative and err on the side of providing more information rather than not quite enough in the case of a command aborting. I'm on the fence on which is the correct behavior, tho.